src/main/webapp/js/instructorFeedbackResults.js

```diff
@@ -77,25 +77,90 @@ function showHideStats(){
     }
 }
 
-//Search functionality
-
+// Filter functionality
+// Filtering is done by searching the heading text in all panels (section, team name, student name)
+//
+// When the heading text of a panel has found to be matched with the search text, 
+// all nested panels in the panel will be shown and parent panels of the panel will be shown as well
+//
+// When at least one of the nested panels in a panel has found to be matched with the search text,
```

Comment:
When at least one of the nested panels in a panel is found to contain the search text in its title,